tests/unit/introspection.tcl

```diff
@@ -533,14 +533,33 @@ start_server {tags {"introspection"}} {
     } {} {external:skip}
 
     test {redis-server command line arguments - save with empty input} {
-        # Take `--loglevel` as the save option value.
-        catch {exec src/redis-server --save --loglevel verbose} err
-        assert_match {*'save "--loglevel" "verbose"'*Invalid save parameters*} $err
+        start_server {config "default.conf" args {--save --loglevel verbose}} {
+            assert_match [r config get save] {save {}}
+            assert_match [r config get loglevel] {loglevel verbose}
+        }
+
+        start_server {config "default.conf" args {--loglevel verbose --save}} {
+            # --save is the last arg, in this case we won't reset the save params
+            assert_match [r config get save] {save {900 1}}
+            assert_match [r config get loglevel] {loglevel verbose}
```

Comment:
are you sure this was the old behavior?
i just tested it on 6.2.7 and it did work (redis started without an empty save config)

Reply:
> i just tested it on 6.2.7 and it did work (redis started without an empty save config)

i am kind of confusing about the wording... i am guessing you did not modify the save config in redis.conf? my steps:
1. 6.2.7
2. git diff
```
-# save 3600 1
+save 3600 1
```
3. src/redis-server redis.conf --save
4. src/redis-cli config get save  ->> 3600 1           the step 3 empty save did not reset the save
5. if we change step 3 to --save "", the step 4 will get the empty save params

i thinks it is a bug in a wrong usage, we should fix it together... both in 7.0 and 6.2

Reply:
i take a look in 6.2.7, if we want to fix it (--save), we also need to check it the argc == 1, like `else if (argc == 1) resetServerSaveParams`
https://github.com/redis/redis/blob/e6f67092f8d4d81761a60c46011d1ff1dc3c2628/src/config.c#L467-L485

Reply:
Let me summarize the changes in a new comment.. there is a mess